crates/ruff_python_semantic/src/model.rs

```diff
@@ -1648,6 +1648,12 @@ impl<'a> SemanticModel<'a> {
         self.flags.intersects(SemanticModelFlags::DOCSTRING)
     }
 
+    /// Return `true` if the model is in an attribute docstring.
+    pub const fn in_attribute_docstring(&self) -> bool {
+        self.flags
+            .intersects(SemanticModelFlags::ATTRIBUTE_DOCSTRING)
+    }
```

Comment:
Should the `in_docstring()` function immediately above this one have its name changed to `in_non_attribute_docstring()`?

Reply:
I guess `in_runtime_stored_docstring` could be used

Reply:
Or alternatively, rather than having two functions, we could have one `docstring_state()` method that returns an enum:

```rs
enum DocstringState {
    ClassOrModule,
    Attribute,
    None,
}
```

This might also better reflect the fact that it's invalid to have both the `ATTRIBUTE_DOCSTRING` and `DOCSTRING` flags simultaneously set. We could add a `debug_assert` in the `docstring_state()` method to enforce that invariant.

Reply:
I would prefer `in_pep_257_docstring` and `in_attribute_docstring`.